Alicerce/Sources/Persistence/SiblingContextCoreDataStack.swift

```diff
@@ -0,0 +1,128 @@
+                                            shouldAddStoreAsynchronously: shouldAddStoreAsynchronously,
+                                            shouldMigrateStoreAutomatically: shouldMigrateStoreAutomatically,
+                                            shouldInferMappingModelAutomatically: shouldInferMappingModelAutomatically,
+                                            storeLoadCompletionHandler: { (store, error) in
+                                                storeLoadCompletionHandler(store + " (work)", error)
+                })
+
+            workContext = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
+            workContext.persistentStoreCoordinator = workCoordinator
+            workContext.mergePolicy = mergePolicy
+
+            let backgroundCoordinator = NestedContextCoreDataStack
+                .persistentStoreCoordinator(withType: storeType,
+                                            storeName: storeName,
+                                            managedObjectModel: managedObjectModel,
+                                            shouldAddStoreAsynchronously: shouldAddStoreAsynchronously,
+                                            shouldMigrateStoreAutomatically: shouldMigrateStoreAutomatically,
+                                            shouldInferMappingModelAutomatically: shouldInferMappingModelAutomatically,
+                                            storeLoadCompletionHandler: { (store, error) in
+                                                storeLoadCompletionHandler(store + " (background)", error)
+                })
+
+            backgroundContext = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+            backgroundContext.persistentStoreCoordinator = backgroundCoordinator
+            backgroundContext.mergePolicy = mergePolicy
+
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(workContextChanged),
+                                                   name: .NSManagedObjectContextDidSave,
+                                                   object: workContext)
+
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(backgroundContextChanged),
+                                                   name: .NSManagedObjectContextDidSave,
+                                                   object: backgroundContext)
+        }
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    private static let defaultStoreLoadCompletionHandler: (Any, Error?) -> Void = { (store, error) in
+        if let error = error {
+            fatalError("💥: Failed to load persistent store \(store)! Error: \(error)")
+        }
+    }
+
+    func context(withType type: CoreDataStackContextType) -> NSManagedObjectContext {
```

Comment:
I had it like that before, but then found it weird when invoking and changed it to its current form. Examples:

`let context = coreDataStack.context(type: .work)`

vs 

`let context = coreDataStack.context(withType: .work)`

I find it more "readable" using the `withType` parameter 😛